src/python/py/models/quantized_model.py

```diff
@@ -52,51 +52,86 @@ def __init__(self):
+        self.scaling = {"default": 1}
+
+
+class QuantizedAttention:
+    def __init__(self, bits, group_size, is_lora):
+        self.q_proj = QuantizedTensorModule(bits, group_size) if not is_lora else QuantizedLoRAModule(bits, group_size)
+        self.k_proj = QuantizedTensorModule(bits, group_size) if not is_lora else QuantizedLoRAModule(bits, group_size)
+        self.v_proj = QuantizedTensorModule(bits, group_size) if not is_lora else QuantizedLoRAModule(bits, group_size)
+        self.o_proj = QuantizedTensorModule(bits, group_size) if not is_lora else QuantizedLoRAModule(bits, group_size)
         self.rotary_emb = TensorModule()
 
 
 class QuantizedMLP:
-    def __init__(self, bits, group_size):
-        self.gate_proj = QuantizedTensorModule(bits, group_size)
-        self.up_proj = QuantizedTensorModule(bits, group_size)
-        self.down_proj = QuantizedTensorModule(bits, group_size)
-        self.fc1 = QuantizedTensorModule(bits, group_size)
-        self.fc2 = QuantizedTensorModule(bits, group_size)
+    def __init__(self, bits, group_size, is_lora):
+        self.gate_proj = QuantizedTensorModule(bits, group_size) if not is_lora else QuantizedLoRAModule(bits, group_size)
+        self.up_proj = QuantizedTensorModule(bits, group_size) if not is_lora else QuantizedLoRAModule(bits, group_size)
+        self.down_proj = QuantizedTensorModule(bits, group_size) if not is_lora else QuantizedLoRAModule(bits, group_size)
+        self.fc1 = QuantizedTensorModule(bits, group_size) if not is_lora else QuantizedLoRAModule(bits, group_size)
+        self.fc2 = QuantizedTensorModule(bits, group_size) if not is_lora else QuantizedLoRAModule(bits, group_size)
 
 
 class QuantizedDecoderLayer:
-    def __init__(self, layer_id, bits, group_size):
+    def __init__(self, layer_id, bits, group_size, is_lora):
         self.layer_id = layer_id
         self.input_layernorm = TensorModule()
-        self.self_attn = QuantizedAttention(bits, group_size)
+        self.self_attn = QuantizedAttention(bits, group_size, is_lora)
         self.post_attention_layernorm = TensorModule()
-        self.mlp = QuantizedMLP(bits, group_size)
+        self.mlp = QuantizedMLP(bits, group_size, is_lora)
 
     def is_empty(self):
         return self.input_layernorm.weight is None
 
 
 class QuantizedModel:
-    def __init__(self, quant_type, input_path, bits, group_size, q_size, kv_size, intermediate_size, num_layers):
+    def __init__(self, quant_type, input_path, bits, group_size, q_size, kv_size, intermediate_size, num_layers, adapter_path):
         self.quant_type = quant_type
         self.embedding = TensorModule()
         self.final_norm = TensorModule()
         self.lm_head = TensorModule()
         self.layers = {}
         self.num_layers = num_layers
 
+        is_lora = adapter_path is not None
+
+        self.map_to_modules(quant_type, input_path, bits, group_size, q_size, kv_size, intermediate_size, adapter_path)
```

Comment:
I did not understand this comment. What is not needed? the is_lora condition or the map_to_modules call?

Reply:
Is `adapter_path` needed in the first call to `self.map_to_modules` since there is a second call to `self.map_to_modules` for `adapter_path`? Can this look instead like the following?

```python
self.map_to_modules(quant_type, input_path, bits, group_size, q_size, kv_size, intermediate_size)
if self.is_lora:
    self.map_to_modules(quant_type, adapter_path, bits, group_size, q_size, kv_size, intermediate_size)
```

Reply:
Not needed since we use self.is_lora now